cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -350,7 +433,7 @@ func translateStubDomainOfKubeDNSToForwardCoreDNS(dataField string, kubeDNSConfi
 // in the form of Proxy for the CoreDNS Corefile.
 func translateUpstreamNameServerOfKubeDNSToUpstreamForwardCoreDNS(dataField string, kubeDNSConfigMap *v1.ConfigMap) (string, error) {
 	if kubeDNSConfigMap == nil {
```

Comment:
This change seems unrelated to the corefile migration.
What is the rational behind this change?

Reply:
When there is no `kube-dns ConfigMap`, the function should return `/etc/resolv.conf`. Else, the CoreDNS Corefile will break with no resolver to the `forward` plugin.
I found this bug while manually testing the upgrades. Probably didn't cause an issue previously because the CoreDNS ConfigMap was always retained and not updated when it already existed.

Reply:
I wonder if this can cause issues with `systemd-resolved` enabled systems (like recent Ubuntu versions)? Have you tried  this @rajansandeep ?

Reply:
@rosti, IIUC, kubeadm already detects `systemd-resolved` systems, and configures `kubelet` to use an alternate `resolv.conf`.  From the perspective of Pods however, the path to this alternate `resolv.conf` is always `/etc/resolv.conf`, regardless of the original path.